tests/test_co2eq_parameters.py

```diff
@@ -384,6 +383,28 @@ class CO2eqParametersDirect(CO2eqParametersDirectAndLifecycleMixin, unittest.Tes
 
     parameters = CO2EQ_PARAMETERS_DIRECT
 
+    # Expected min and max values for emission factors, by mode.
+    ranges_by_mode = {
```

Comment:
```suggestion
    ranges_by_mode: Dict[str, Tuple[Union[int, float], Union[int, float]]] = {
```
You might have to update the imports from typing after adding this.

Reply:
Done; note that I discovered that type `float` sort of includes `int`; and there's also type numbers.Number which is already imported that includes float, int and other number types.

https://stackoverflow.com/questions/50928592/mypy-type-hint-unionfloat-int-is-there-a-number-type